network_importer/adapters/netbox_api/models.py

```diff
@@ -256,15 +256,23 @@ def delete(self) -> Optional["DiffSyncModel"]:
             NetboxInterface: DiffSync object
         """
         # Check if the interface has some Ips, check if it is the management interface
+
         if self.ips:
-            dev = self.diffsync.get(self.diffsync.device, identifier=self.device_name)
-            if dev.primary_ip and dev.primary_ip in self.ips:
-                LOGGER.warning(
-                    "Unable to delete interface %s on %s, because it's currently the management interface",
-                    self.name,
-                    dev.name,
+            try:
+                dev = self.diffsync.get(self.diffsync.device, identifier=self.device_name)
+                if dev.primary_ip and dev.primary_ip in self.ips:
+                    LOGGER.warning(
+                        "Unable to delete interface %s on %s, because it's currently the management interface",
+                        self.name,
+                        dev.name,
+                    )
+                    return self
+            except ObjectNotFound:
+                LOGGER.error(
+                    "Unable to delete interface %s on %s, because device is not present.", self.name, self.device_name,
```

Comment:
This might be hiding a bigger problem, do you know why the device is not present anymore ? that shouldn't happen as far as I know